lib/AST/ASTImporter.cpp

```diff
@@ -7786,14 +7747,38 @@ Decl *ASTImporter::Import(Decl *FromD) {
     return ToD;
   }
 
-  // Import the type.
+  // Import the declaration.
   ExpectedDecl ToDOrErr = Importer.Visit(FromD);
   if (!ToDOrErr) {
-    llvm::consumeError(ToDOrErr.takeError());
-    return nullptr;
+    // Failed to import.
+    // Take out the existing (probably invalid) object from the mapping.
+    // FIXME: There may be remaining references to the failed object.
```

Comment:
The failed Decl should be removed from lookup table and ImportedFromDecls too.

Reply:
We should not forget this, could you please create an issue for that?

Reply:
In the last commit this problem was corrected. (Remaining references in AST can still be problem but this was the case previously too.)